app/src/main/java/sw19/moring03/paint/tools/CircleTool.java

```diff
@@ -36,6 +38,8 @@ public boolean draw(Canvas canvas, Paint paint) {
 
         radius = (float)Math.sqrt(Math.pow((double)Math.abs(center.x - outerPoint.x), (double)2) + Math.pow((double)Math.abs(center.y - outerPoint.y), (double)2));
 
+        paint.setPathEffect(pathEffect);
```

Comment:
I would also set the pathEffect property of the paint in DrawingView.onDraw(), so we stay consistent. Color and stroke width is also set there.